File: src/CSharpCore/DreamEngine/Sync/GameSynchronizer.cs
using DreamEngine.Sync.Data;

namespace DreamEngine.Sync;
internal static class GameSynchronizer
{
    public static unsafe void Synchronize(ref GameData gameData, EntityData* entities, int entityCount)
    {
        Game.DeltaTime = gameData.deltaTime;
        InputSynchronizer.Synchronize(ref gameData.inputData);
        SceneSynchronizer.SynchronizeFromTo(ref gameData.sceneData, Game.Scene, entities, entityCount);
    }
}

File: src/Core/Sync/GameSynchronizer.cpp
#include "GameSynchronizer.h"

#include "Application.h"
#include "InputSynchronizer.h"

using namespace DreamEngine::Core::Sync;

GameData* GameSynchronizer::Synchronize(const bool isFocused)
{
    GameData* gameData = Application::Instance().GetGameData();
    gameData->deltaTime = Application::Instance().GetDeltaTime();
    gameData->sceneData = *Application::Instance().GetGame()->GetActiveScene()->GetSceneData();

    if (isFocused)
        gameData->inputData = *InputSynchronizer::Synchronize();

    return gameData;
}
